examples/userdemo/connect_simple01.py

```diff
@@ -3,12 +3,16 @@
 Connect Simple01
 ================
 
+A `ConnectionPatch` can be used to draw a line (possibly with arrow head)
```

Comment:
This should be in text and annotations, not burried in the catch-all `userdemo` section (which as far as I'm concerned shouldn't exist).

Reply:
The `userdemo` section contains the examples from some tutorials, in this case from the [annotations tutorial](https://matplotlib.org/tutorials/text/annotations.html).  
It might be a good idea to disintegrate this section, but as usual the problem is with breaking links. So *if* there is consensus on doing so, I would propose to do it all at once.

Reply:
This new functionality should get a demo in the proper place not the catch all. Of course you can have both if you want the new functionality in the old demo as well.

Reply:
I feel there should be one single `ConnectionPatch` demo/example. The functionality added here is just an extention to the existing `coords*` arguments, which makes this consistent with other artists like the FancyArrowPatch, which also allows for custom coordinate systems (but not in different axes).